server/src/embedded-languages/documents-manager.ts

```diff
@@ -47,12 +41,12 @@ export default class EmbeddedDocumentsManager {
     embeddedDocumentContent: string,
     partialEmbeddedDocumentInfos: Omit<EmbeddedDocumentInfos, 'uri'>
   ): void {
-    const originalPath = originalUriString.replace('file://', '')
-    const pathToEmbeddedDocumentsFolder = this.getPathToEmbeddedDocumentFolder(originalPath)
-    fs.mkdirSync(pathToEmbeddedDocumentsFolder, { recursive: true })
+    const randomName = randomUUID()
```

Comment:
Just to be sure, you always use the same random file for each language document, or is a new random file generated each parsing?